src/dmd/parse.d

```diff
@@ -2868,7 +2868,14 @@ final class Parser(AST) : Lexer
                         // Don't call nextToken again.
                     }
                 case TOK.in_:
-                    stc = STC.in_;
+                    if (global.params.inMeansScopeConst)
+                    {
+                        // `in` now means `const scope` as originally intented
+                        stc = STC.const_ | STC.scope_;
```

Comment:
I don't think that's the right fix. It means there is a discrepancy between header / JSON generation when this preview switch is activated.

Before:
```D
alias Dg = void delegate(in char[]) @safe;

void stringify(scope Dg sink) @safe {
    sink("oops");
}

void main() @safe {
    auto x = bar();
}

const(char)[] bar() @safe {
    string str;
    const(char)[] arg;
    stringify((chars) { arg ~= chars; });
    return null;
}
```

Turns into:
```D
// D import file generated from 'atila.d'
alias Dg = void delegate(in char[]) @safe;
@safe void stringify(scope Dg sink);
@safe void main();
@safe const(char)[] bar();
```

With this switch on:
```D
// D import file generated from 'atila.d'
alias Dg = void delegate(scope const char[]) @safe;
@safe void stringify(scope Dg sink);
@safe void main();
@safe const(char)[] bar();
```

Reply:
What's the problem in practice?

Reply:
Sorry it took this long to answer...

The problem is that `in` is  used in multiple places in DMD. It is its own storage class, which is weird, but the way it is currently.

Just substituting it in the parse phase is not enough. grep for `STC.in_`, you'll see it used for calling , you'll see it added to variadic arguments for D-style variadic functions (https://github.com/dlang/dmd/blob/91adbef4bc92f01908673719a476b779c0eb5bf8/src/dmd/expressionsem.d#L2332), used for array iteration (https://github.com/dlang/dmd/blob/91adbef4bc92f01908673719a476b779c0eb5bf8/src/dmd/statementsem.d#L1811), etc...

I did a few tests and since those are mostly druntime calls, and `scope` is ignored, most seems harmless (although they could interfere with escape analysis), but those kinds of disparities will come back to bite us at some point.